### ISSUE TITLE
Support Swagger EvilDiscriminator for OneOf used by Microsoft.Insights

### ISSUE THREAD

**Issue by matthchr** [Azure/azure-service-operator#2004]
The spec for this resource is derived from [this](https://schema.management.azure.com/schemas/2018-03-01/Microsoft.Insights.json#/definitions/MetricAlertCriteria) and ends up looking like:

```
//Generated from: https://schema.management.azure.com/schemas/2018-03-01/Microsoft.Insights.json#/definitions/MetricAlertCriteria
type MetricAlertCriteria struct {
	//MetricAlertMultipleResourceMultipleMetric: Mutually exclusive with all other
	//properties
	MetricAlertMultipleResourceMultipleMetric *MetricAlertCriteria_MetricAlertMultipleResourceMultipleMetric `json:"metricAlertMultipleResourceMultipleMetricCriteria,omitempty"`

	//MetricAlertSingleResourceMultipleMetric: Mutually exclusive with all other
	//properties
	MetricAlertSingleResourceMultipleMetric *MetricAlertCriteria_MetricAlertSingleResourceMultipleMetric `json:"metricAlertSingleResourceMultipleMetricCriteria,omitempty"`

	//WebtestLocationAvailability: Mutually exclusive with all other properties
	WebtestLocationAvailability *MetricAlertCriteria_WebtestLocationAvailability `json:"webtestLocationAvailabilityCriteria,omitempty"`
}
```

whereas the Status is derived from [the Swagger](https://github.com/Azure/azure-rest-api-specs/blob/767f6ae9f6345f1e466c469e3c51bd7df4c3e0ee/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json#L732) and looks like this:
```
type MetricAlertCriteria_Status struct {
	// +kubebuilder:validation:Required
	//OdataType: specifies the type of the alert criteria.
	OdataType MetricAlertCriteriaStatusOdataType `json:"odata.type"`
}
```

**Comment by matthchr** [comment]
Will this _also_ be fixed by @porges work on Swagger?

**Comment by matthchr** [comment]
This will be resolved by #2323

**Comment by theunrepentantgeek** [comment]
Confirmed as fixed in the Swagger WIP branch (see #2323).

The Spec version looks like this:

``` go
type MetricAlertCriteria struct {
	// MicrosoftAzureMonitorMultipleResourceMultipleMetric: Mutually exclusive with all other properties
	MicrosoftAzureMonitorMultipleResourceMultipleMetric *MetricAlertMultipleResourceMultipleMetricCriteria `json:"microsoftAzureMonitorMultipleResourceMultipleMetricCriteria,omitempty"`
	// MicrosoftAzureMonitorSingleResourceMultipleMetric: Mutually exclusive with all other properties
	MicrosoftAzureMonitorSingleResourceMultipleMetric *MetricAlertSingleResourceMultipleMetricCriteria `json:"microsoftAzureMonitorSingleResourceMultipleMetricCriteria,omitempty"`
	// MicrosoftAzureMonitorWebtestLocationAvailability: Mutually exclusive with all other properties
	MicrosoftAzureMonitorWebtestLocationAvailability *WebtestLocationAvailabilityCriteria `json:"microsoftAzureMonitorWebtestLocationAvailabilityCriteria,omitempty"`
}
```

and the Status version looks like this:
``` go
type MetricAlertCriteria_STATUS struct {
	// MicrosoftAzureMonitorMultipleResourceMultipleMetric: Mutually exclusive with all other properties
	MicrosoftAzureMonitorMultipleResourceMultipleMetric *MetricAlertMultipleResourceMultipleMetricCriteria_STATUS `json:"microsoftAzureMonitorMultipleResourceMultipleMetricCriteria,omitempty"`
	// MicrosoftAzureMonitorSingleResourceMultipleMetric: Mutually exclusive with all other properties
	MicrosoftAzureMonitorSingleResourceMultipleMetric *MetricAlertSingleResourceMultipleMetricCriteria_STATUS `json:"microsoftAzureMonitorSingleResourceMultipleMetricCriteria,omitempty"`
	// MicrosoftAzureMonitorWebtestLocationAvailability: Mutually exclusive with all other properties
	MicrosoftAzureMonitorWebtestLocationAvailability *WebtestLocationAvailabilityCriteria_STATUS `json:"microsoftAzureMonitorWebtestLocationAvailabilityCriteria,omitempty"`
}
```
